types/express-serve-static-core/index.d.ts

```diff
@@ -440,7 +440,10 @@ interface Request extends http.IncomingMessage, Express.Request {
 
     method: string;
 
-    params: any;
+    params: {
+        [namedParameter: string]: string;
+        [captureGroup: number]: string;
+    };
```

Comment:
@19majkel94 Numeric lookups would still work without the the numeric key signature, presumably because TypeScript knows the key will be coerced into a string:

``` ts
{
    const params: {
        [namedParameter: string]: string;
        // [captureGroup: number]: string;
    } = {};

    const v2 = params.foo;
    const v1 = params[1]; // no error
}
```